numba_dppy/tests/test_dpnp_functions.py

```diff
@@ -267,6 +264,732 @@ def f(a):
+        set_dpnp_debug(None)
+
+    def test_ranf(self):
+        from numba.tests.support import captured_stdout
+
+        @njit
+        def f(size):
+            c = np.random.ranf(size)
+            return c
+
+        sizes = [9, (2, 5), (3, 2, 4)]
+        set_dpnp_debug(1)
+        with captured_stdout() as got_gpu_message:
+            with dpctl.device_context("opencl:gpu"):
+                for size in sizes:
+                    result = f(size)
+                    _result = result.ravel()
+                    for i in range(_result.size):
+                        self.assertTrue(_result[i] >= 0.0)
+                        self.assertTrue(_result[i] < 1.0)
+                    self.assertIn("DPNP implementation", got_gpu_message.getvalue())
+        set_dpnp_debug(None)
+
+    def test_sample(self):
+        from numba.tests.support import captured_stdout
+
+        @njit
+        def f(size):
+            c = np.random.sample(size)
+            return c
+
+        sizes = [9, (2, 5), (3, 2, 4)]
+        set_dpnp_debug(1)
+        with captured_stdout() as got_gpu_message:
+            with dpctl.device_context("opencl:gpu"):
+                for size in sizes:
+                    result = f(size)
+                    _result = result.ravel()
+                    for i in range(_result.size):
+                        self.assertTrue(_result[i] >= 0.0)
+                        self.assertTrue(_result[i] < 1.0)
+                    self.assertIn("DPNP implementation", got_gpu_message.getvalue())
+        set_dpnp_debug(None)
+
+    def test_random(self):
+        from numba.tests.support import captured_stdout
+
+        @njit
+        def f(size):
+            c = np.random.random(size)
+            return c
+
+        sizes = [9, (2, 5), (3, 2, 4)]
+        set_dpnp_debug(1)
+        with captured_stdout() as got_gpu_message:
+            with dpctl.device_context("opencl:gpu"):
+                for size in sizes:
+                    result = f(size)
+                    _result = result.ravel()
+                    for i in range(_result.size):
+                        self.assertTrue(_result[i] >= 0.0)
+                        self.assertTrue(_result[i] < 1.0)
+                    self.assertIn("DPNP implementation", got_gpu_message.getvalue())
+        set_dpnp_debug(None)
+
+    def test_rand(self):
+        from numba.tests.support import captured_stdout
+
+        @njit
+        def f():
+            c = np.random.rand(3, 2)
+            return c
+
+        set_dpnp_debug(1)
+        with captured_stdout() as got_gpu_message:
+            with dpctl.device_context("opencl:gpu"):
+                result = f()
+                _result = result.ravel()
+                for i in range(_result.size):
+                    self.assertTrue(_result[i] >= 0.0)
+                    self.assertTrue(_result[i] < 1.0)
+                self.assertIn("DPNP implementation", got_gpu_message.getvalue())
```

Comment:
I think we should move checks outside the device context. It will show explicitly which one functions we need to call inside device context.

Reply:
Sounds good.